GoogleForCreators/web-stories-wp
# Storybook: New Story Details Modal

## Context

The new story details modal that shows when a user clicks 'publish' is going to be _alot_ of code to review so I'm doing it in steps to get buy in along the way and keep review time shorter and easier to see details. 

## Summary

This is the initial scaffolding of the [new modal](https://www.figma.com/file/ZrO763yvMWJjYeAPsRnNfW/_Editor---Sprint-Feature-Level-Designs_2022?node-id=57%3A14792). It's got a different header and action pattern than other modals so I didn't reuse the design system's `Dialog` for it. 

This update is only visible in the new Storybook Component I made (lives here: /stories-editor-components-dialog-publish-modal--default). 

Here's what it'll look like from Figma
<img width="810" alt="Screen Shot 2022-01-27 at 4 57 29 PM" src="https://user-images.githubusercontent.com/10720454/151463030-9db8c5d7-885b-49ed-ba85-8536397c84a3.png">


## Relevant Technical Choices

Just some initial file structure to try and keep file size small and establish patterning as well as initial foundational styling (lookin' at you, css grid). 

Gave grid percentage sizing so dialog can grow for massive screens while respecting figma as the minimum/proper ratio. 

Decided to name the grid sections by their contents in a feeble and lofty attempt at readability for future versions of ourselves tracking down file paths, let me know if it's excessive.  Current HTML structure will also ensure proper tabbing order when that becomes relevant (checklist should be last thing tabbed to not first thing from cover preview button). 

All of the content here is clearly dummy. Added purely to make visualizing what I did easier.  I also didn't worry about scroll content since the paneling will add its own complexity. 

## To-do

- all functionality 
- ~follow up with Amy on if close button should be square or circle on hover/focus~
- ~follow up with Amy about if the 1px line between panels and header is intentional~

## User-facing changes

Nothing

## Testing Instructions

Spin up storybook locally and see this under stories editor/components/ dialog/publish modal

<img width="956" alt="Screen Shot 2022-01-28 at 8 13 23 AM" src="https://user-images.githubusercontent.com/10720454/151571835-6f2b7181-fd92-4a7a-9640-8b1218c88259.png">






This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Partially addresses #10115
